### PR TITLE
flatten multipart transfers

### PR DESCRIPTION
**Pull request by whyrusleeping**
This is based on #2039 so wait until it merges to review this.

But this removes the whole recursive multipart thing.

The multipart 'Next()' bug in #1688 is what i set out to fix, but that was apparently fixed in go version 1.5.2

### CODE REVIEW

**Comment by jbenet** [comment]
#2039 is merged

File: commands/files/file_test.go [hunk 1]
```diff
@@ -86,23 +88,9 @@ Some-Header: beep
 
 beep
 --Boundary!
-Content-Type: multipart/mixed; boundary=OtherBoundary
+Content-Type: application/x-directory
```
Comment (Dignifiedquire) [inline commands/files/file_test.go]: Should test file inside the directory as well

File: commands/files/multipartfile.go [hunk 2]
```diff
@@ -12,7 +12,8 @@ const (
 	multipartFormdataType = "multipart/form-data"
 	multipartMixedType    = "multipart/mixed"
 
-	applicationSymlink = "application/symlink"
+	applicationDirectory = "application/x-directory"
+	applicationSymlink   = "application/symlink"
```
Comment (Dignifiedquire) [inline commands/files/multipartfile.go]: Is this tested and documented anywhere (symlink)?

File: commands/files/serialfile.go [hunk 3]
```diff
@@ -68,6 +70,15 @@ func (f *serialFile) NextFile() (File, error) {
 	stat := f.files[0]
 	f.files = f.files[1:]
 
+	for !f.hidden && strings.HasPrefix(stat.Name(), ".") {
```
Comment (Dignifiedquire) [inline commands/files/serialfile.go]: Probably best to rename `.hidden` to something like `.addHidden` or `.ignoreHidden`

**Comment by whyrusleeping** [comment]
cool, addressed your concerns @Dignifiedquire merging.

**Pull request merged**